internal/pyxis/pyxis.go

```diff
@@ -0,0 +1,80 @@
+package pyxis
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/shurcooL/graphql"
+)
+
+const (
+	DefaultPyxisHost = "catalog.redhat.com/api/containers"
+)
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type PyxisClient struct {
+	Client    HTTPClient
```

Comment:
If we're going to remove the interface definition, you might even embed the HTTP client here.

Reply:
I switched to just use an `http.Client` here directly.